test/functional/hooks.js

```diff
@@ -41,38 +42,122 @@ describe('library hooks', () => {
           'dynamicName', dynamicValue);
     }
 
+    function noop() {}
+
+    function stubOut(mutator) {
+      stub = sandbox.stub();
+      mutators.attributes[mutator] = stub;
+    }
+
     beforeEach(() => {
-      sandbox.spy(attributes, 'applyAttr');
+      allSpy = sandbox.spy(mutators.attributes, '__all');
     });
-  
-    it('should be called for static attributes', () => {
-      patch(container, render, 'dynamicValue');
-      var el = container.childNodes[0];
 
-      expect(attributes.applyAttr).calledWith(el, 'staticName', 'staticValue');
+    afterEach(() => {
+      for (var mutator in mutators.attributes) {
+        if (mutator !== '__all') {
+          mutators.attributes[mutator] = null;
+        }
+      }
     });
 
-    it('should be called for dynamic attributes', () => {
-      patch(container, render, 'dynamicValue');
-      var el = container.childNodes[0];
 
-      expect(attributes.applyAttr).calledWith(el, 'dynamicName', 'dynamicValue');
+    describe('for static attributes', () => {
+      it('should call specific setter', () => {
+        stubOut('staticName');
+
+        patch(container, render, 'dynamicValue');
+        var el = container.childNodes[0];
+
+        expect(stub).calledOnce;
```

Comment:
I was just following the style [already there](https://github.com/google/incremental-dom/pull/92/files#diff-cd7507808e5cf41a0923565b24b06e33L59). :wink:

Updated.